pkgs/applications/audio/sonobus/default.nix

```
@@ -0,0 +1,83 @@
+{ lib
+, pkg-config
+, stdenv
+, fetchFromGitHub
+, alsa-lib
+, cmake
+, fetchzip
+, freetype
+, libGL
+, libX11
+, libXcursor
+, libXext
+, libXinerama
+, libXrandr
+, libjack2
+, libopus
+, opusTools
+, curl
+, gtk3
+, webkitgtk
+}:
+
+stdenv.mkDerivation rec {
+  pname = "sonobus";
+  version = "1.6.0";
+
+  src = fetchFromGitHub {
+    owner = "sonosaurus";
+    repo = "sonobus";
+    rev = version;
+    sha256 = "sha256-Qz5nUgT/p6BuO1VGz5PiDxvv/1ctBOTVrG4BAbJS9/g=";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+  ];
+
+  buildInputs = [
+    alsa-lib
+    freetype
+    libGL
+    libX11
+    libXcursor
+    libXext
+    libXinerama
+    libXrandr
+    libjack2
+    libopus
+    opusTools
+    curl
+    gtk3
+    webkitgtk
+  ];
+
+  # To add the libraries to the rpath, needed or the program segfaults
+  NIX_LDFLAGS = "-rpath ${lib.makeLibraryPath buildInputs}";
+  dontPatchELF = true;
+
+  postPatch = lib.optionalString (stdenv.isLinux) ''
+    # needs special setup on Linux, dunno if it can work on Darwin
+    # https://github.com/NixOS/nixpkgs/issues/19098
+    # Also, I get issues with linking without that, not sure why
+    sed -i -e '/juce::juce_recommended_lto_flags/d' CMakeLists.txt
+    patchShebangs linux/install.sh
+  '';
+
+  # The program does not provide any CMake install instructions
+  installPhase = lib.optionalString (stdenv.isLinux) ''
+    cd ../linux
+    ./install.sh "$out"
```

```suggestion
    runHook preInstall
    cd ../linux
    ./install.sh "$out"
    runHook postInstall
```

Don't forget to run hooks if you set a custom phase